plugins/http2_adapter/lib/src/connection_manager_imp.dart

```diff
@@ -175,7 +180,7 @@ class _ConnectionManager implements ConnectionManager {
         context: clientConfig.context,
         onBadCertificate: clientConfig.onBadCertificate,
         supportedProtocols: ['h2'],
-      );
+      ).timeout(_handshakeTimout);
```

Comment:
Do we have other libraries that have a similar approach? I'm not sure how this is related to *handshake* only. Could it be related to other processes?

Reply:
The only example I have in mind that allows to explicitly set handhshake timeout is `net/http.Transport` package in Go that has a property `TLSHandshakeTimeout`

In Java there's a [`Socket.connect`](https://developer.android.com/reference/java/net/Socket#connect(java.net.SocketAddress,%20int)) method that allows to set TCP-connection timeout which is also applied to handshake because it can only begin after connection is established

Technically in this case this property is applied only to [`handshake future`](https://github.com/dart-lang/sdk/blob/9f2218e4509210576abecc79d7184c94a3ad0e26/sdk/lib/io/secure_socket.dart#L633)